packages/headless/src/controllers/commerce/recommendations/headless-recommendations.ts

```diff
@@ -0,0 +1,116 @@
+import {recommendationsOptionsSchema} from '../../../features/commerce/recommendations/recommendations';
+import {
+  fetchRecommendations,
+  updateRecommendationsSlotId,
+} from '../../../features/commerce/recommendations/recommendations-actions';
+import {recommendationsReducer as recommendations} from '../../../features/commerce/recommendations/recommendations-slice';
+import {loadReducerError} from '../../../utils/errors';
+import {validateInitialState} from '../../../utils/validate-payload';
+import {
+  buildController,
+  Controller,
+} from '../../controller/headless-controller';
+
+/**
+ * The `Recommendation` controller exposes a method for retrieving recomendations content in a commerce interface.
+ */
+export interface Recommendations extends Controller {
+  /**
+   * Fetches the recommendations.
+   */
+  refresh(): void;
+
+  /**
+   * A scoped and simplified part of the headless state that is relevant to the `Recommendation` controller.
+   */
+  state: RecommendationsState;
+}
+
+export interface RecommendationsState {
+  headline: string;
+  products: Product[];
+  error: CommerceAPIErrorStatusResponse | null;
+  isLoading: boolean;
+  responseId: string;
+}
+
+export interface RecommendationsOptions {
+  slotId: string;
+}
+
+interface RecommendationsProps {
+  options: RecommendationsOptions;
+}
+
+function validateRecommendationInitialState(
+  engine: CommerceEngine,
+  options: RecommendationsOptions
+) {
+  if (!options) {
+    return;
+  }
+
+  validateInitialState(
+    engine,
+    recommendationsOptionsSchema,
+    options,
+    'buildRecommendations'
+  );
+}
+/**
+ * Creates a `Recommendations` controller instance.
+ *
+ * @param engine - The headless commerce engine.
+ * @returns A `Recommendation` controller instance.
+ */
+export function buildRecommendations(
+  engine: CommerceEngine,
+  props: RecommendationsProps
+): Recommendations {
+  if (!loadBaseRecommendationsReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+
+  validateRecommendationInitialState(engine, props.options);
+  dispatch(updateRecommendationsSlotId({slotId: props.options.slotId}));
+
+  const recommendationStateSelector = createSelector(
+    (state: CommerceEngineState) => state.recommendations,
+    (recommendation) => ({
+      headline: recommendation.headline,
+      products: recommendation.products,
+      error: recommendation.error,
+      isLoading: recommendation.isLoading,
+      responseId: recommendation.responseId,
+    })
+  );
+
+  return {
+    ...controller,
+
+    get state() {
+      return recommendationStateSelector(engine.state);
+    },
+
+    refresh: () => dispatch(fetchRecommendations()),
+  };
+}
+
+function loadBaseRecommendationsReducers(
+  engine: CommerceEngine
+): engine is CommerceEngine {
+  engine.addReducers({recommendations, commerceContext, configuration});
```

Comment:
My understanding is that since these reducers will have been loaded by now, there's no need to re-load them.
```suggestion
  engine.addReducers({recommendations});
```

Reply:
The test that checks they have been loaded if I remove them, so I don't think they are being loaded

Reply:
What I mean is that to use a recs controller, one will have loaded a context controller before, which loads the context reducer: https://github.com/coveo/ui-kit/blob/fe2a772d3b61aac90fb407fd494769e9b03f6683/packages/headless/src/controllers/commerce/context/headless-context.ts#L168-L173

I thought the config controller was loaded in the context controller as well but it isn't. Nonetheless, this means we could remove the context reducer (and adjust the assertion in the test as well).
```suggestion
  engine.addReducers({recommendations, commerceContext});
```